### PR TITLE
support subdirectory

### PR DESCRIPTION
**Pull request by maz1987in**
Fix the issue of 404 if the application is in a subdirectory

### CODE REVIEW

**Comment by maz1987in** [comment]
@amol-  could you please look at this fix?

File: depot/middleware.py [hunk 1]
```diff
@@ -185,9 +185,12 @@ def __call__(self, environ, start_response):
                 full_path[mtpointlen:mtpointlen+1] in ('', '/')):
             return self.app(environ, start_response)
 
+        
+        mountpointlen=len(self.mountpoint.split('/'))
         path = full_path.split('/')
```
Comment (amol-) [inline depot/middleware.py:190]: I think that changing this `full_path.split('/')` to be `full_path.rsplit('/', 2)`  is the only change that we need to support subdirectories

**Comment by amol-** [comment]
This PR should also add a test to confirm that it works as expected and prevent regressions. https://github.com/amol-/depot/blob/master/tests/test_wsgi_middleware.py is probably the right place for it

**Comment by maz1987in** [comment]
@amol-  Done adding the test case and tuning the code

**Review by maz1987in — COMMENTED** [review]
tuning the code thanks @amol-

**Pull request merged**